app/src/main/java/com/example/mytestproject/ui/weatherData/dailyWeather/TenDaysWeatherFragment.kt

```diff
@@ -28,20 +31,19 @@ class TenDaysWeatherFragment : Fragment() {
         return inflater.inflate(R.layout.fragment_ten_days_weather, container, false)
     }
 
-    override fun onActivityCreated(savedInstanceState: Bundle?) {
-        super.onActivityCreated(savedInstanceState)
-
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
         (activity?.applicationContext as App).weatherDataComponent.inject(this)
 
-        setTitle("Moscow")
+        setTitle(loadCityName(view.context))
```

Comment:
City name should be got in ViewModel and passed to Fragment to display in title. 
In ViewModel you get weather data. Maybe you can extract name of the city from this data? 
If not, let's create a separate UseCase for that, like a GetCityDetails.
Atfer that we can get rid of saving filtered city data to preferences (in saveFilteredCityList).

Reply:
Fixed